### PR TITLE
Allow users to mount local volumes

### PR DESCRIPTION
**Pull request by faiq**
allows users to mount local volumes through a `Volumes` section in their wercker.yml file

in order to run this, users must use a `--enable-volumes` flag

we leverage this feature to test wercker inside of a wercker build container as well

### CODE REVIEW

**Comment by faiq** [comment]
for original POC see #49

File: test-all.sh [hunk 1]
```diff
@@ -66,10 +66,9 @@ testDirectMount() {
 
 runTests() {
   basicTest "source-path" build $testsDir/source-path || return 1
-  basicTest "test local services" build $testsDir/local-service/service-consumer || return 1
+  basicTest "test local services" --debug build  $testsDir/local-service/service-consumer || return 1
```
Comment (termie) [inline test-all.sh]: this one doesn't need debug when it is default now

File: wercker.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
-box: tcnksm/gox:1.5.1
+box:
+  id: tcnksm/gox:1.5.1
+  volumes: $CODE_PATH /var/run/docker.sock
```
Comment (termie) [inline wercker.yml]: please make this box only for the test pipeline (copy this to there and then remove the volumes from this one)

**Comment by termie** [comment]
looks good, small changes before merge

**Comment by termie** [comment]
.. and rebase probably

File: docker/box.go [hunk 3]
```diff
@@ -168,6 +169,19 @@ func (b *DockerBox) binds() ([]string, error) {
...
+	if b.options.EnableVolumes {
+		vols := util.SplitSpaceOrComma(b.config.Volumes)
+		var interpolatedVols []string
+		for _, vol := range vols {
+			interpolatedVols = append(interpolatedVols, env.Interpolate(vol))
+		}
+		b.volumes = interpolatedVols
+	}
+
+	for _, volume := range b.volumes {
```
Comment (hatchan) [inline docker/box.go]: We can move this inside the if statement, right?

File: test-all.sh [hunk 4]
```diff
@@ -84,10 +84,9 @@ testScratchPush () {
 
 runTests() {
   basicTest "source-path" build $testsDir/source-path || return 1
-  basicTest "test local services" build $testsDir/local-service/service-consumer || return 1
+  basicTest "test local services" build  $testsDir/local-service/service-consumer || return 1
```
Comment (hatchan) [inline test-all.sh]: The extra space is not needed

**Comment by hatchan** [comment]
Is is possible to mount certain container paths to other paths on the host? Something like:

```
/some_path:/other_path
```

**Comment by hatchan** [comment]
Is it also possible to use the volumes for services? That would be really neat, as that would make it possible to have persistent databases between tests.

**Comment by hatchan** [comment]
Minor comments, looks to me :+1:

File: docker/box.go (hunk 3 above)
Comment (faiq) [inline docker/box.go]: yea we can.

**Comment by faiq** [comment]
> Is is possible to mount certain container paths to other paths on the host? Something like:

> /some_path:/other_path

as of right now, its not. we need to use maps in wercker.yml in order to support this, its not a priority for me right now but its not a hard fix i think. 


> Is it also possible to use the volumes for services? That would be really neat, as that would make it possible to have persistent databases between tests.

I think so. mostly because services are based on top of boxes (assuming you have the `--enable-volumes` flag on you should be able to do it. but not too sure. tapping in @termie for more words.

anyways fixed the changes and will merge.

**Comment by hatchan** [comment]
>> Is is possible to mount certain container paths to other paths on the host? Something like:
>> 
>> /some_path:/other_path

> as of right now, its not. we need to use maps in wercker.yml in order to support this, its not a priority for me right now but its not a hard fix i think.

Not necessary, we can just use the same notation that I've used above. Which I believe is the same as the one that Docker uses. 

```
volumes: /home/ /host_path/:/container_path/
```
In this example `/home/` from the host will be in the container on path `/home/`. And `/host_path/` will be mounted on `/container_path/`.

**Comment by Tonnu** [comment]
@faiq did you write docs for this by any chance?

**Comment by faiq** [comment]
@Tonnu nope net yet :/

**Pull request merged**